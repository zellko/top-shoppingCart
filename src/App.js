import React, { useState } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Home from './components/Home';
import Shop from './components/Shop';
import Cart from './components/Cart';
import Nav from './components/Nav';
import CartHook from './components/CartHook';
import './styles/App.css';

function App() {
  const {
    cart,
    addItemToCart,
    removeItemFromCart,
    updateItemQty,
  } = CartHook();

  return (
    <BrowserRouter>
      <div className="App">
        <Nav cartLength={cart.length} />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/shop" element={<Shop onAddToCart={addItemToCart} />} />
          <Route
            path="/cart"
            element={(
              <Cart
                cart={cart}
                onRemoveItem={removeItemFromCart}
                onUpdateQuantity={updateItemQty}
              />
            )}
          />
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
