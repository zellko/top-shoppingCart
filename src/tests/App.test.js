import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import App from '../App';

describe('App component test', () => {
  it('render Home page', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('render Shop page', async () => {
    const { container } = render(<App />);

    const homeButton = screen.getByRole('button', { name: 'Go Shopping' });
    expect(homeButton).toBeInTheDocument();

    userEvent.click(homeButton);

    await waitFor(() => screen.getByRole('heading', { name: 'PRODUCT' }));

    const shopHeading = screen.getByRole('heading', { name: 'PRODUCT' });

    expect(shopHeading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('render Cart page', async () => {
    const { container } = render(<App />);

    const cartButton = screen.getAllByRole('link')[2];
    expect(cartButton).toBeInTheDocument();

    userEvent.click(cartButton);

    await waitFor(() => screen.getByRole('heading', { name: 'Your shopping cart is empty' }));

    const cartHeading = screen.getByRole('heading', { name: 'Your shopping cart is empty' });

    expect(cartHeading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
