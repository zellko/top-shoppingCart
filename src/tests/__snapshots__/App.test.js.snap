// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test render Cart page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="nav"
    >
      <ul>
        <a
          href="/"
        >
          <div
            class="logo"
          >
            <div
              class="triangle"
            />
            <div
              class="circle"
            />
            <div
              class="square"
            />
          </div>
        </a>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
        <a
          href="/cart"
        >
          <li>
            Cart
            <span>
              0
            </span>
          </li>
        </a>
      </ul>
    </div>
    <div
      class="cart"
    >
      <div
        class="cart-header"
      >
        <h1>
          Your shopping cart is empty
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component test render Home page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="nav"
    >
      <ul>
        <a
          href="/"
        >
          <div
            class="logo"
          >
            <div
              class="triangle"
            />
            <div
              class="circle"
            />
            <div
              class="square"
            />
          </div>
        </a>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
        <a
          href="/cart"
        >
          <li>
            Cart
            <span>
              0
            </span>
          </li>
        </a>
      </ul>
    </div>
    <div
      class="home"
    >
      <div
        class="home-logo"
      >
        <div
          class="triangle"
        />
        <div
          class="circle"
        />
        <div
          class="square"
        />
      </div>
      <h1>
        ShapeShop
      </h1>
      <a
        href="/shop"
      >
        <button>
          Go Shopping
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`App component test render Shop page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="nav"
    >
      <ul>
        <a
          href="/"
        >
          <div
            class="logo"
          >
            <div
              class="triangle"
            />
            <div
              class="circle"
            />
            <div
              class="square"
            />
          </div>
        </a>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
        <a
          href="/cart"
        >
          <li>
            Cart
            <span>
              0
            </span>
          </li>
        </a>
      </ul>
    </div>
    <div
      class="shop"
    >
      <div
        class="shop-header"
      >
        <h3>
          PRODUCT
        </h3>
        <h3>
          PRICE
        </h3>
      </div>
      <h3>
        Loading...
      </h3>
      <div />
    </div>
  </div>
</div>
`;
